src/main/drivers/rx_nrf24l01.c

```diff
@@ -0,0 +1,346 @@
+
+    // Bit 0 goes to RF_DR_HIGH, bit 1 - to RF_DR_LOW
+    rf_setup = (rf_setup & 0xD7) | ((bitrate & 0x02) << 4) | ((bitrate & 0x01) << 3);
+    return NRF24L01_WriteReg(NRF24L01_06_RF_SETUP, rf_setup);
+}
+
+uint8_t NRF24L01_SetPower(uint8_t power)
+{
+    if(power > 3)
+        power = 3;
+    // Power is in range 0..3 for nRF24L01
+    rf_setup = (rf_setup & 0xF9) | ((power & 0x03) << 1);
+    return NRF24L01_WriteReg(NRF24L01_06_RF_SETUP, rf_setup);
+}
+
+void NRF24L01_SetTxRxMode(enum TXRX_State mode)
+{
+    if(mode == TX_EN) {
+        NRF24_CE_LO();
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, (1 << NRF24L01_07_STATUS_RX_DR)    //reset the flag(s)
+                                            | (1 << NRF24L01_07_STATUS_TX_DS)
+                                            | (1 << NRF24L01_07_STATUS_MAX_RT));
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)   // switch to TX mode
+                                            | (1 << NRF24L01_00_CONFIG_CRCO)
+                                            | (1 << NRF24L01_00_CONFIG_PWR_UP));
+        delayMicroseconds(130);
+        NRF24_CE_HI();
+    } else if (mode == RX_EN) {
+        NRF24_CE_LO();
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, 0x70);        // reset the flag(s)
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, 0x0F);        // switch to RX mode
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, (1 << NRF24L01_07_STATUS_RX_DR)    //reset the flag(s)
+                                            | (1 << NRF24L01_07_STATUS_TX_DS)
+                                            | (1 << NRF24L01_07_STATUS_MAX_RT));
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)   // switch to RX mode
+                                            | (1 << NRF24L01_00_CONFIG_CRCO)
+                                            | (1 << NRF24L01_00_CONFIG_PWR_UP)
+                                            | (1 << NRF24L01_00_CONFIG_PRIM_RX));
+        delayMicroseconds(130);
+        NRF24_CE_HI();
+    } else {
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)); //PowerDown
+        NRF24_CE_LO();
+    }
+}
+
+bool NRF24L01_Reset()
+{
+    NRF24L01_SetTxRxMode(TXRX_OFF);
+    return NRF24L01_ReadReg(0x07) == 0x0E;
+}
+
+/*
+ * Enter standby mode
+ */
+void NRF24L01_SetStandbyMode(void)
+{
+    // set CE low and clear the PRIM_RX bit to enter standby mode
+    NRF24_CE_LO();
+    NRF24L01_WriteReg(NRF24L01_00_CONFIG, standbyConfig);
+}
+
+/*
+ * Enter receive mode
+ */
+void NRF24L01_SetRxMode(void)
+{
+    NRF24_CE_LO(); // drop into standby mode
+    // set the PRIM_RX bit
+    NRF24L01_WriteReg(NRF24L01_00_CONFIG, standbyConfig | BV(NRF24L01_00_CONFIG_PRIM_RX));
+    NRF24L01_ClearAllInterrupts();
+    // finally set CE high to start enter RX mode
+    NRF24_CE_HI();
+    // nRF24L01+ needs 130 microseconds settling time from Standby mode to RX mode
+    delayMicroseconds(130);
```

Comment:
This is better - but there should be no reason to wait here - reception will be started automatically and first packet will arrive much later anyway ..